numpy/distutils/fcompiler/gnu.py

```diff
@@ -126,8 +126,15 @@ def get_flags_linker_so(self):
                 # from it.
                 import distutils.sysconfig as sc
                 g = {}
-                filename = sc.get_makefile_filename()
-                sc.parse_makefile(filename, g)
+                try:
+                    filename = sc.get_makefile_filename()
+                    sc.parse_makefile(filename, g)
+                except AttributeError as e:
+                    if 'get_makefile_filename' in e.message:
```

Comment:
Instead of checking the error message, would using an `else` block work? e.g.,
```
try:
    get_filename = sc.get_makefile_filename
except AttributeError:
    pass  # no Makefile
else:
    sc.parsefile(get_filename(), g)
```

It's generally a good idea to contain no more than the single line at which the exception can occur in the `try` block.